### PR TITLE
feat(debugapi, postage): dilute batch handling

### PR DESCRIPTION
**Pull request by aloknerurkar**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2410)

### CODE REVIEW

File: pkg/debugapi/postage.go [hunk 1]
```diff
@@ -381,3 +379,57 @@ func (s *Service) postageTopUpHandler(w http.ResponseWriter, r *http.Request) {
 		BatchID: id,
 	})
 }
+
+func (s *Service) postageDiluteHandler(w http.ResponseWriter, r *http.Request) {
+	idStr := mux.Vars(r)["id"]
+	if idStr == "" || len(idStr) != 64 {
```
Comment (notanatol) [inline pkg/debugapi/postage.go:385]: I think the first check is redundant, if it's empty string it will fail the len check anyway

Comment (aloknerurkar) [reply pkg/debugapi/postage.go:385]: This is how we do it in other API endpoints as well. The reasoning behind it, AFAIK:

Empty string check ` x == ""` is far more efficient than `len()` check. If in the `||` the first condition is true, the second one is not evaluated at all. This is why I would prefer having this check and evaluating empty strings slightly early.

**Review by acud — DISMISSED** [review]
very nice. I have some small requests 

> Reviewed 16 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2410#-:-MhJ82-_3N1BMv38cO_1:b-nkavh8)* status: all files reviewed, 5 unresolved discussions (waiting on @aloknerurkar, @notanatol, and @zelig)

___
*[openapi/SwarmDebug.yaml, line 924 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-5iwCWdy3tswtts0:-MhJ-5iwCWdy3tswtts1:bafp9uy) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/openapi/SwarmDebug.yaml#L924)):*
> ```YAML
>             $ref: "SwarmCommon.yaml#/components/schemas/BatchID"
>           required: true
>           description: Swarm address of the stamp
> ```

batch ID to dilute
___
*[pkg/debugapi/postage.go, line 385 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhDVGkHDPV3H4wkmnsr-r1-385:-MhJ-CAM0E4q6WdO9UD7:b22db8m) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage.go#L385)):*
<details><summary><i>Previously, aloknerurkar wrote…</i></summary><blockquote>

This is how we do it in other API endpoints as well. The reasoning behind it, AFAIK:

Empty string check ` x == ""` is far more efficient than `len()` check. If in the `||` the first condition is true, the second one is not evaluated at all. This is why I would prefer having this check and evaluating empty strings slightly early.
</blockquote></details>

if `x == ""` the router will not hit the endpoint and return an error since this parameter is mandatory. so `idStr` will never be `""`. If this is on other endpoints then this is a mistake that needs to be rectified
___
*[pkg/debugapi/postage_test.go, line 115 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-_Ay8urGtz3mkWP0:-MhJ-_Ay8urGtz3mkWP1:bdi81ht) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage_test.go#L115)):*
> ```Go
> 
> 	t.Run("with too many requests error", func(t *testing.T) {
> 		wait, done := make(chan struct{}), make(chan struct{})
> ```

since you're testing the middleware here it might be useful to just have one test for that (since this test is copied between different endpoints) maybe it would be easier to just pull this test out and run it on both endpoints, instead of copying it
___
*[pkg/debugapi/postage_test.go, line 485 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-hKQ8P3acSgNMTNc:-MhJ-hKQ8P3acSgNMTNd:b-kps1la) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage_test.go#L485)):*
> ```Go
> 	t.Run("with too many requests error", func(t *testing.T) {
> 		wait, done := make(chan struct{}), make(chan struct{})
> 		contract := contractMock.New(
> ```

see ^^
___
*[pkg/debugapi/postage_test.go, line 655 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-khE1_87kzY7JKg1:-MhJ-khE1_87kzY7JKg2:b-kps1la) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage_test.go#L655)):*
> ```Go
> 	t.Run("with too many requests error", func(t *testing.T) {
> 		wait, done := make(chan struct{}), make(chan struct{})
> 		contract := contractMock.New(
> ```

see ^^

**Review by zelig — APPROVED** [review]
> Reviewed 16 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2410)* status: all files reviewed, 5 unresolved discussions (waiting on @aloknerurkar and @notanatol)

**Review by aloknerurkar — COMMENTED** [review]
> Dismissed @acud and @notanatol from 5 discussions.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2410)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @aloknerurkar)

___
*[openapi/SwarmDebug.yaml, line 924 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-5iwCWdy3tswtts0:-MhTdWKlDDm2aJyBj-JY:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/openapi/SwarmDebug.yaml#L924)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

batch ID to dilute
</blockquote></details>

Done.
___
*[pkg/debugapi/postage.go, line 385 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhDVGkHDPV3H4wkmnsr-r1-385:-MhTdaJ-12K5eZ2AqR2Q:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage.go#L385)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

if `x == ""` the router will not hit the endpoint and return an error since this parameter is mandatory. so `idStr` will never be `""`. If this is on other endpoints then this is a mistake that needs to be rectified
</blockquote></details>

Done.
___
*[pkg/debugapi/postage_test.go, line 115 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-_Ay8urGtz3mkWP0:-MhTddpF4OMAQTO_UeGt:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage_test.go#L115)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

since you're testing the middleware here it might be useful to just have one test for that (since this test is copied between different endpoints) maybe it would be easier to just pull this test out and run it on both endpoints, instead of copying it
</blockquote></details>

Done.
___
*[pkg/debugapi/postage_test.go, line 485 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-hKQ8P3acSgNMTNc:-MhTderi7-o0Ah7h2kS8:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage_test.go#L485)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

see ^^
</blockquote></details>

Done.
___
*[pkg/debugapi/postage_test.go, line 655 at r1](https://reviewable.io/reviews/ethersphere/bee/2410#-MhJ-khE1_87kzY7JKg1:-MhTdgCS3ygGJH7h_cOy:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/bf5eb27d9037f41ffcb229a5beabb4ac0d7912ff/pkg/debugapi/postage_test.go#L655)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

see ^^
</blockquote></details>

Done.

**Review by acud — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2410#-:-MhmL-I7DLoZRvC27Zql:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @aloknerurkar)

**Pull request merged**